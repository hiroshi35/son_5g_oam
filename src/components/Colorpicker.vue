<template>
  <div>
    <div v-if="icon" class='input-group' ref="element">
      <input type='text' class="form-control" :placeholder="placeholder" :value="value">
      <span class="input-group-addon"><i></i></span>
    </div>
    <input v-else type="text" class="form-control" :placeholder="placeholder" ref="element" :value="value">
  </div>

</template>
<script>
  import 'bootstrap-colorpicker'
  export default {
    props: {
      placeholder: {
        type: String,
        default: 'Enter Color'
      },
      options: {
        type: Object,
        default () {
          return {
            format: 'hex'
          }
        }
      },
      icon: {
        type: Boolean,
        default: false
      },
      value: {
        type: String
      }
    },
    watch: {
      color (val) {
        this.$emit('input', val)
      },
      value (val) {
        $(this.$refs.element).colorpicker('setValue', val)
      }
    },
    mounted () {
      let vm = this
      $(this.$refs.element).colorpicker(vm.options)
      $(this.$refs.element).colorpicker().on('changeColor', function (e) {
        vm.color = e.color.toHex()
      })
    },
    data () {
      return {
        color: ''
      }
    }
  }
</script>

<style lang="scss">
  @import "~bootstrap-colorpicker/dist/css/bootstrap-colorpicker";
</style>

