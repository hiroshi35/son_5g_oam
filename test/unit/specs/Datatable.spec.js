import { shallow } from 'vue-test-utils'
import Colorpicker from '@/components/Datatable.vue'
import {datatable} from 'jquery'

describe('VSelect2.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Colorpicker, {
      propsData: {
        columns: [{title: 'Define'}, {title: 'Your'}, {title: 'Column Names'}, {title: 'Here'}],
        tableData: [['First', 'Second', 'Third', 'Fourth']]
      }
    })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  test('$().DataTable() should be called', () => {
    expect(datatable).toBeCalled()
  })
  it('has correct props', () => {
    expect(wrapper.props().columns).toEqual([{title: 'Define'}, {title: 'Your'}, {title: 'Column Names'}, {title: 'Here'}])
    expect(wrapper.props().tableData).toEqual([['First', 'Second', 'Third', 'Fourth']])
  })
})
