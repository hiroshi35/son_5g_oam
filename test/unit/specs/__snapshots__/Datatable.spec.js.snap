// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSelect2.vue has the expected html structure 1`] = `
<div>
  <table
    class="table table-bordered"
  />
   
  <!---->
</div>
`;
